### DISCUSSION TITLE
Why doesn't the placeholder option on choice/entity form types work with multiple=true?

### DISCUSSION THREAD

**Discussion by jerrac** [symfony/symfony#49209] [Q&A] [answered]
Per the docs, "This option only applies if the multiple option is set to false."

I'm honestly confused on this one. I have a form I'm working on right now where I need a way to set "no value". But the Choice/EntityType doesn't give me that option unless I don't allow selecting multiple items. 

I mean, removing data is a normal operation. Why would a form allowing you to select multiple options somehow make that not true? I'm super confused on this one....

**Comment by MatTheCat** [comment]
When you can select multiple values, “no value” is the same thing as not selecting any value so there is no need for a placeholder.

**Reply by jerrac** [reply]
When you edit something, and values are already selected, how do you deselect all values when you need to remove all values for some reason?

**Reply by norkunas** [reply]
Ctrl+click

**Reply by jerrac** [reply]
@norkunas Thanks. That helps. It's both obvious, and not obvious at the same time. A developer like me didn't think of it, but when you think about how you select multiple options, it suddenly seems obvious. I actually just implemented a button that will deselect everything to make the ui a bit easier in my app.

**Comment by stof** [comment, marked as answer]
This is done this way because this is the way HTML works.
For a field supporting multiple values, submitted an empty value would not be the same than submitting no value at all (`count([''])` is `1`, not `0`)

**Reply by stof** [reply]
see https://html.spec.whatwg.org/multipage/form-elements.html#placeholder-label-option for the concept of `placeholder label option` in the HTML spec. You'll see that it only applies to non-multiple options (the spec also makes it apply only for required selects, treating it like a normal choice for other cases, but that's mostly because it would not change anything for a non-required select)

**Reply by jerrac** [reply]
So the placeholder is an empty string?

**Reply by stof** [reply]
In a select, the placeholder label option is the option that has an empty string as value.
If you submit such form (disabling the frontend validation so that `required` does not block the submission), the empty string is indeed the submitted value.

Btw, trying to add a "placeholder" option in a `<select multiple>` would actually break also the form validation for `required` when you select that option as it make the selection non empty.

**Reply by jerrac** [reply]
Ok. That explains things. Thanks. :)
